packages/jest-resolve/src/fileWalkers.ts

```diff
@@ -30,7 +30,8 @@ function statSyncCached(path: string): IPathType {
 
   let stat;
   try {
-    stat = fs.statSync(path);
+    // @ts-expect-error TS2554 - throwIfNoEntry is only available in recent version of node, but inclusion of the option is a backward compatible no-op.
+    stat = fs.statSync(path, {throwIfNoEntry: false});
```

Comment:
can we remove the `try-catch` now? Seeing as all we check is that the error is not "not found"? At least the `ENOENT` condition can be removed?

Reply:
try/catch is still required for older node versions. in those versions, this new option will be ignored, and an error will need to be thrown and caught, just as before. Can only remove try/catch once node@14.17 is the lowest supported version

Reply:
Right, of course 👍